swagger.yaml

```
@@ -44,21 +422,255 @@ paths:
+            readOnly: true
+            example: "2025-04-04T07:00:00.000Z"
+          id:
+            type: string
+            description: "The bundle id"
+            readOnly: true
+            example: "9e4e3628-fc85-48cd-80ad-e005d9d283ff"
+          last_updated_by:
+            description: "The user that last updated the bundle, taken from the auth token supplied."
+            type: string
+            example: "publisher@ons.gov.uk"
+          preview_teams:
+            description: "A list of teams who have permissions to view the dataset series in the bundle.  Supplied by AWS cognito."
+            type: array
+            items:
+              type: string
+          publish_date_time:
+            description: "The time and date of publication."
+            type: string
+            format: date-time
+            example: "2025-04-04T07:00:00.000Z"
+          state:
+            description: "The approval status of the overall bundle (TBC - values provided are examples only)."
+            type: string
+            enum:
+              - not_started
+              - in_progress
+              - approved
+            example: "approved"
+          title:
+            type: string
+            description: "The title of the bundle"
+            example: "CPI March 2025"
+          updated_date:
+            description: "The time and date the bundle was last updated."
+            type: string
+            readOnly: true
+            format: date-time
+            example: "2025-04-04T07:00:00.000Z"
+          wagtail_managed:
+            description: "If the bundle creation was initiated by wagtail"
+            type: boolean
+  Contents:
+    description: "A list of contents related to a bundle"
+    type: object
+    allOf:
+      - $ref: "#/definitions/PaginationFields"
+      - type: object
+        properties:
+          items:
+            type: array
+            items:
+              $ref: "#/definitions/ContentItem"
+  ContentItem:
+    description: "A model which holds information about the datasets to be published as part of the bundle"
+    allOf:
+      - type: object
+        properties:
+          dataset_id:
+            type: string
+            description: "The dataset id of the item in the bundle"
+            example: "cpih"
+          edition_id:
+            type: string
+            description: "The edition id of the dataset item in the bundle"
+            example: "march"
+          item_id:
+            type: string
+            readOnly: true
+            description: "An auto generated id field to identify the item"
+            example: "DE3BC0B6-D6C4-4E20-917E-95D7EA8C91DC"
+          state:
+            description: "The approval status of the item, hydrated from dataset-api when this item is retrieved from the database."
+            type: string
+            readOnly: true
+            enum:
+              - associated
+              - published
+            example: "published"
```

Consider workflow state that are relevant to the context of the bundle workflow here. Dataset states could then be mapped to these generic states to abstract away any further complexity from those integrating to the bundles API. This would also make it easier to extend with other content types without breaking changes or inconsistency.

```suggestion
              - PENDING
              - APPROVED
              - RELEASED
            example: APPROVED
```

If review and approval of individual items is introduced then additional states may be needed.

---

This has been amended